pre-receive-hooks/block_self_merge_prs.sh

```diff
@@ -8,10 +8,9 @@
 # https://help.github.com/enterprise/admin/guides/developer-workflow/managing-pre-receive-hooks-on-the-github-enterprise-appliance/
 #
 
-if [ "$GITHUB_VIA" = "merge api" && "$GITHUB_PULL_REQUEST_AUTHOR_LOGIN" = "$GITHUB_USER_LOGIN"]; then
+if [[ "$GITHUB_VIA" = "merge api" ]] && [[ "$GITHUB_PULL_REQUEST_AUTHOR_LOGIN" = "$GITHUB_USER_LOGIN" ]]; then
```

Comment:
@nathos: merged this for now to fix the syntax issues but I am all in to add additional methods when to block in a follow up PR. I talked to engineering on how to block direct git push and got the advise to combine with another hook which would block direct pushes of all but some white listed users.